cmd/worker/main.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/go-logr/logr"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/cmd"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/worker"
+	"github.com/spf13/cobra"
+	"k8s.io/klog/v2"
+	"k8s.io/klog/v2/klogr"
+)
+
+var (
+	configHelper = worker.NewConfigHelper()
+	logger       logr.Logger
+	w            worker.Worker
+)
+
+var rootCmd = &cobra.Command{
+	Use:               "worker",
+	CompletionOptions: cobra.CompletionOptions{DisableDefaultCmd: true},
+}
+
+var kmodCmd = &cobra.Command{
+	Use:   "kmod",
+	Short: "Manage kernel modules",
+}
+
+var kmodLoadCmd = &cobra.Command{
+	Use:   "load",
+	Short: "Load a kernel module",
+	Args:  cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cfgPath := args[0]
+
+		logger.V(1).Info("Reading config", "path", cfgPath)
+
+		cfg, err := configHelper.ReadConfigFile(cfgPath)
+		if err != nil {
+			return fmt.Errorf("could not read config file %s: %v", cfgPath, err)
+		}
+
+		return w.LoadKmod(cmd.Context(), cfg)
+	},
+}
+
+var kmodReloadCmd = &cobra.Command{
+	Use:   "reload",
+	Short: "Unload a kernel module and load it again with a different configuration",
+	Args:  cobra.ExactArgs(2),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		oldCfgPath := args[0]
+
+		logger.V(1).Info("Reading old config", "path", oldCfgPath)
+
+		oldCfg, err := configHelper.ReadConfigFile(oldCfgPath)
+		if err != nil {
+			return fmt.Errorf("could not read config file %s: %v", oldCfgPath, err)
+		}
+
+		newCfgPath := args[1]
+
+		logger.V(1).Info("Reading new config", "path", newCfgPath)
+
+		newCfg, err := configHelper.ReadConfigFile(args[1])
+		if err != nil {
+			return fmt.Errorf("could not read config file %s: %v", newCfgPath, err)
+		}
+
+		return w.ReloadKmod(cmd.Context(), oldCfg, newCfg)
+	},
+}
+
+var kmodUnloadCmd = &cobra.Command{
+	Use:   "unload",
+	Short: "Unload a kernel module",
+	Args:  cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cfgPath := args[0]
+
+		logger.V(1).Info("Reading config", "path", cfgPath)
```

Comment:
It is not exactly the same in all 3 commands, because `kmod reload` needs to read for the old and the new config; what we log there is slightly different. @yevgeny-shnaidman WDYT?

Reply:
I think we should leave as is - the more explicit the logs, the easier the debugging :)